mypy/errors.py

```diff
@@ -553,12 +563,11 @@ def is_errors(self) -> bool:
 
     def is_real_errors(self) -> bool:
         """Are there any generated errors (not just notes, for example)?"""
-        return any(info.severity == 'error'
-                   for infos in self.error_info_map.values() for info in infos)
+        return self.has_real_errors
 
     def is_blockers(self) -> bool:
         """Are the any errors that are blockers?"""
-        return any(err for errs in self.error_info_map.values() for err in errs if err.blocker)
+        return self.has_blockers
```

Comment:
This one makes sense to me. We check for blockers after processing every single file, so with a lot of errors that naturally gets quadratic.

Reply:
Add to change from a single boolean to a path-indexed Set to accommodate removal of errors